python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+
+                sed_name = '%s/specFileGLSN_%i_%i_%.4f.txt.gz' % (sne_file_loc,
+                                                                  spr_sys_df['twinkles_system'].iloc[0],
+                                                                  image_on, visit_mjd)
+
+                if (spr_sys_df['sedFilepath'].iloc[idx] != sed_name):
+                    if sed_error is False:
+                        errors_string = errors_string + "\nSNe Image SED Filename. First error found in gal_id: %i " % u_id
+                        errors_present = True
+                        sed_error = True
+        
+        print('------------')
+        print('SNe Images direct catalog input Results:')
+
+        if errors_present is False:
+            print('Pass: All instance catalog values within 0.005 of lensed system inputs. ' +
+                  'All SED Filenames match.')
+        else:
+            raise CatalogError('\nFail:\n%s' % errors_string)
+
+        print('------------')
+
+        return
+
+    def compare_agn_image_mags(self, spr_agn_df, spr_agn_lens_df, visit_mjd,
+                               visit_band):
+
+        """
+        This test compares the image magnitudes.
+        """
+        
+        db = om10.DB(catalog=os.path.join(os.environ['TWINKLES_DIR'], 'data', 
+                                          'twinkles_lenses_v2.fits'), vb=False)
+
+        agnSpecDir = 'agnSED'
+        agnDir = str(getPackageDir('sims_sed_library') + '/' + agnSpecDir)
+        bandpassDict = BandpassDict.loadTotalBandpassesFromFiles(bandpassNames=[visit_band])
+        norm_bp = Bandpass()
+        norm_bp.imsimBandpass()
+
+        agn_var_params = self.load_agn_var_params()
+
+        max_magNorm_err = 0.
+
+        for lens_gal_row in spr_agn_lens_df.iterrows():
+
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            spr_sys_df = spr_agn_df.query('lens_galaxy_uID == %i' % u_id)
+            agn_id = spr_sys_df['galaxy_id']
+            use_lens = db.lenses['LENSID'][np.where(db.lenses['twinklesId'] == 
+                                                    spr_sys_df['twinkles_system'].iloc[0])[0]]
+            lens = db.get_lens(use_lens)
+            num_img = lens['NIMG']
+
+            mag = lens['MAG'].data[0]
+            lensed_mags = spr_sys_df['phosimMagNorm']
+            corrected_mags = []
+
+            for i in range(num_img.data[0]):
+
+                d_mags = self.get_agn_variability_mags(agn_var_params[str(agn_id.values[0])],
+                                                       lens['DELAY'].data[0][i], visit_mjd,
+                                                       spr_sys_df['redshift'].values[0])
+
+
+                test_sed = Sed()
+                test_sed.readSED_flambda('%s/%s' % (agnDir, 'agn.spec.gz'))
+                test_sed.redshiftSED(lens['ZSRC'])
+                test_f_norm = test_sed.calcFluxNorm(lensed_mags.values[i], norm_bp)
```

Comment:
The method `lsst.sims.photUtils.getImsimFluxNorm` is generally faster than calling `calcFluxNorm` with the imsim bandpass.  I recommend using that here.